packages/ai-providers/server-ai-langchain/src/ldai_langchain/langchain_provider.py

```diff
@@ -0,0 +1,273 @@
+
+            # Generate metrics early (assumes success by default)
+            metrics = LangChainProvider.get_ai_metrics_from_response(response)
+
+            # Extract text content from the response
+            content: str = ''
+            if isinstance(response.content, str):
+                content = response.content
+            else:
+                # Log warning for non-string content (likely multimodal)
+                if self.logger:
+                    self.logger.warn(
+                        f'Multimodal response not supported, expecting a string. '
+                        f'Content type: {type(response.content)}, Content: {response.content}'
+                    )
+                # Update metrics to reflect content loss
+                metrics = LDAIMetrics(success=False, usage=metrics.usage)
+
+            # Create the assistant message
+            assistant_message = LDMessage(role='assistant', content=content)
+
+            return ChatResponse(
+                message=assistant_message,
+                metrics=metrics,
+            )
+        except Exception as error:
+            if self.logger:
+                self.logger.warn(f'LangChain model invocation failed: {error}')
+
+            return ChatResponse(
+                message=LDMessage(role='assistant', content=''),
+                metrics=LDAIMetrics(success=False, usage=None),
+            )
+
+    async def invoke_structured_model(
+        self,
+        messages: List[LDMessage],
+        response_structure: Dict[str, Any],
+    ) -> StructuredResponse:
+        """
+        Invoke the LangChain model with structured output support.
+
+        :param messages: Array of LDMessage objects representing the conversation
+        :param response_structure: Dictionary defining the output structure
+        :return: StructuredResponse containing the structured data
+        """
+        try:
+            # Convert LDMessage[] to LangChain messages
+            langchain_messages = LangChainProvider.convert_messages_to_langchain(messages)
+
+            # Get the LangChain response with structured output
+            structured_llm = self._llm.with_structured_output(response_structure)
+            response = await structured_llm.ainvoke(langchain_messages)
+
+            # Using structured output doesn't support metrics
+            metrics = LDAIMetrics(
+                success=True,
+                usage=TokenUsage(total=0, input=0, output=0),
+            )
+
+            # Handle response serialization
+            if isinstance(response, dict):
+                raw_response = str(response)
+            else:
+                import json
+                try:
+                    raw_response = json.dumps(response)
+                except (TypeError, ValueError):
+                    raw_response = str(response)
+
+            return StructuredResponse(
+                data=response if isinstance(response, dict) else {'result': response},
```

Comment:
I am a little concerned that we are creating our own structured response object if we don't get a dict back. The result should be the structured response specified otherwise we should not return return a value.